mllib/src/main/scala/org/apache/spark/ml/clustering/LDA.scala

```diff
@@ -0,0 +1,740 @@
+    "Concentration parameter (commonly named \"beta\" or \"eta\") for the prior placed on topic'" +
+      " distributions over terms.", (beta: Double) => beta == -1 || beta >= 0.0)
+
+  /** @group getParam */
+  @Since("1.6.0")
+  def getTopicConcentration: Double = $(topicConcentration)
+
+  /**
+   * Alias for [[getTopicConcentration]]
+   * @group getParam
+   */
+  @Since("1.6.0")
+  def getBeta: Double = getTopicConcentration
+
+  /**
+   * Optimizer or inference algorithm used to estimate the LDA model, specified as a
+   * [[LDAOptimizer]] type.
+   * Currently supported:
+   *  - Online Variational Bayes: [[OnlineLDAOptimizer]] (default)
+   *  - Expectation-Maximization (EM): [[EMLDAOptimizer]]
+   * @group param
+   */
+  @Since("1.6.0")
+  final val optimizer = new Param[LDAOptimizer](this, "optimizer", "Optimizer or inference" +
+    " algorithm used to estimate the LDA model")
+
+  /** @group getParam */
+  @Since("1.6.0")
+  def getOptimizer: LDAOptimizer = $(optimizer)
+
+  // Developers should override these setOptimizer() methods.  These are defined here to
+  // ensure identical behavior when setting the optimizer using a String.
+  /** @group setParam */
+  @Since("1.6.0")
+  def setOptimizer(value: LDAOptimizer): this.type = set(optimizer, value)
+
+  /**
+   * Set [[optimizer]] by name (case-insensitive):
+   *  - "online" = [[OnlineLDAOptimizer]]
+   *  - "em" = [[EMLDAOptimizer]]
+   * @group setParam
+   */
+  @Since("1.6.0")
+  def setOptimizer(value: String): this.type = value.toLowerCase match {
+    case "online" => setOptimizer(new OnlineLDAOptimizer)
+    case "em" => setOptimizer(new EMLDAOptimizer)
+    case _ => throw new IllegalArgumentException(
+      s"LDA was given unknown optimizer '$value'.  Supported values: em, online")
+  }
+
+  /**
+   * Output column with estimates of the topic mixture distribution for each document (often called
+   * "theta" in the literature).  Returns a vector of zeros for an empty document.
+   *
+   * This uses a variational approximation following Hoffman et al. (2010), where the approximate
+   * distribution is called "gamma."  Technically, this method returns this approximation "gamma"
+   * for each document.
+   * @group param
+   */
+  @Since("1.6.0")
+  final val topicDistributionCol = new Param[String](this, "topicDistribution", "Output column" +
+    " with estimates of the topic mixture distribution for each document (often called \"theta\"" +
+    " in the literature).  Returns a vector of zeros for an empty document.")
+
+  setDefault(topicDistributionCol -> "topicDistribution")
+
+  /** @group getParam */
+  @Since("1.6.0")
+  def getTopicDistributionCol: String = $(topicDistributionCol)
+
+  /**
+   * Validates and transforms the input schema.
+   * @param schema input schema
+   * @return output schema
+   */
+  protected def validateAndTransformSchema(schema: StructType): StructType = {
+    SchemaUtils.checkColumnType(schema, $(featuresCol), new VectorUDT)
```

Comment:
I don't think we should provide the `Vector` API at all if the vector can only contain integer arguments since we really just mean `Array[Int]`. Making the change now would allow us to drop this legacy API altogether. If we need to merge this quickly, can we make the public API support `Array[Int]` and do a conversion to vector internally?